src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -4049,22 +4057,23 @@ def update_vmss(cmd, resource_group_name, name, license_type=None, no_wait=False
         else:
             vmss.virtual_machine_profile.billing_profile.max_price = max_price
 
-    if security_type is not None:
+    if security_type is not None or enable_secure_boot is not None or enable_vtpm is not None:
         security_profile = vmss.virtual_machine_profile.security_profile
-        vmss_security_type = security_profile.security_type if security_profile else None
-        if vmss_security_type != security_type:
+        prev_security_type = security_profile.security_type if security_profile else None
+        # At present, `SecurityType` only has option `TrustedLaunch`
+        if security_type is not None and prev_security_type != security_type:
             vmss.virtual_machine_profile.security_profile = {
                 'securityType': security_type,
                 'uefiSettings': {
                     'secureBootEnabled': enable_secure_boot if enable_secure_boot is not None else False,
                     'vTpmEnabled': enable_vtpm if enable_vtpm is not None else True
                 }
             }
-    elif enable_secure_boot is not None or enable_vtpm is not None:
-        vmss.virtual_machine_profile.security_profile = {'uefiSettings': {
-            'secureBootEnabled': enable_secure_boot,
-            'vTpmEnabled': enable_vtpm
-        }}
+        else:
+            vmss.virtual_machine_profile.security_profile = {'uefiSettings': {
+                'secureBootEnabled': enable_secure_boot,
+                'vTpmEnabled': enable_vtpm
+            }}
```

Comment:
How about setting `uefiSettings` and `securityType` separately, such as 
```
vmss.virtual_machine_profile.security_profile = {'uefiSettings': {
    'secureBootEnabled': enable_secure_boot,
    'vTpmEnabled': enable_vtpm
}}
if security_type is not None and prev_security_type != security_type:
      vmss.virtual_machine_profile.security_profile.securityType = security_type
```

Reply:
> How about setting `uefiSettings` and `securityType` separately, such as
> 
> ```
> vmss.virtual_machine_profile.security_profile = {'uefiSettings': {
>     'secureBootEnabled': enable_secure_boot,
>     'vTpmEnabled': enable_vtpm
> }}
> if security_type is not None and prev_security_type != security_type:
>       vmss.virtual_machine_profile.security_profile.securityType = security_type
> ```
![image](https://github.com/Azure/azure-cli/assets/32201005/1be496e7-65f3-4ca2-b542-98907918f443)
![image](https://github.com/Azure/azure-cli/assets/32201005/74636130-7d4c-4c30-928e-7e49d5552a93)

According to the Scenario 2 and Scenario 3, we should set the default value of `secureBootEnabled` and `vTpmEnabled` if not specified by end user when updating security-type. 
While if end user tries to update `uefiSetting` without update `securityType`, the default value of `secureBootEnabled` and `vTpmEnabled` should be `None`.